src/bin/edit/state.rs

```diff
@@ -42,8 +42,7 @@ pub struct DisplayablePathBuf {
 impl DisplayablePathBuf {
     #[allow(dead_code, reason = "only used on Windows")]
     pub fn from_string(str: String) -> Self {
-        let value = PathBuf::from(&str);
-        Self { value, str: Cow::Owned(str) }
+        Self::from_path(PathBuf::from(str))
```

Comment:
Good catch!

But this will call `to_string_lossy` which we should avoid. Can you please employ the same transmute trick from below to get a `Cow::Borrowed` first and then create the `PathBuf`?